tests/src/main/java/net/neoforged/neoforge/debug/capabilities/handlers/resources/ComponentResourceTests.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.debug.capabilities.handlers.resources;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.level.material.Fluids;
+import net.neoforged.neoforge.capabilities.Capabilities;
+import net.neoforged.neoforge.common.util.FriendlyByteBufUtil;
+import net.neoforged.neoforge.fluids.FluidType;
+import net.neoforged.neoforge.transfer.ResourceHandlerUtil;
+import net.neoforged.neoforge.transfer.TransferAction;
+import net.neoforged.neoforge.transfer.handlers.IItemContext;
+import net.neoforged.neoforge.transfer.handlers.resources.IResourceHandler;
+import net.neoforged.neoforge.transfer.handlers.templates.contexts.PlayerItemContext;
+import net.neoforged.neoforge.transfer.handlers.templates.contexts.StackItemContext;
+import net.neoforged.neoforge.transfer.handlers.templates.fluids.FluidResourceContainerContents;
+import net.neoforged.neoforge.transfer.handlers.templates.fluids.ItemContextFluidHandler;
+import net.neoforged.neoforge.transfer.handlers.templates.items.ItemResourceContainerContents;
+import net.neoforged.neoforge.transfer.handlers.templates.resource.ResourceContainerContents;
+import net.neoforged.neoforge.transfer.resources.FluidResource;
+import net.neoforged.neoforge.transfer.resources.ItemResource;
+import net.neoforged.neoforge.transfer.resources.ResourceStack;
+import net.neoforged.neoforge.transfer.transaction.Transaction;
+import net.neoforged.neoforge.transfer.transaction.TransactionManager;
+import net.neoforged.testframework.annotation.ForEachTest;
+import net.neoforged.testframework.annotation.TestHolder;
+import net.neoforged.testframework.gametest.EmptyTemplate;
+import net.neoforged.testframework.gametest.ExtendedGameTestHelper;
+import net.neoforged.testframework.gametest.GameTest;
+
+// Tests component storages on ItemStacks as well as the handler logic for those components. Also validates the Codec and Stream codec
+@ForEachTest(groups = ResourceHandlerTestSetup.GROUP_ID, idPrefix = "resource.handler.component.")
+public class ComponentResourceTests {
+    @GameTest
+    @EmptyTemplate
+    @TestHolder(description = "Tests that FluidHandlerItemStack works")
```

Comment:
This should more accurately describe the test at hand; looking at the code it should include something like "inserting and draining fluids from a bucket should change the item"